src/BackgroundTasks/Task/Job.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace ILIAS\BackgroundTasks\Task;
+
+use ILIAS\BackgroundTasks\IO;
+use ILIAS\BackgroundTasks\Task;
+
+/**
+ * Interface Job
+ *
+ * @package ILIAS\BackgroundTasks\Task
+ *
+ *          A Task, which can be run without any interaction with the user such as zipping files or just collecting some data
+ */
+interface Job extends Task {
+
+	/**
+	 * @param \ILIAS\BackgroundTasks\IO $input
+	 * @return \ILIAS\BackgroundTasks\IO
+	 */
+	public function run(IO $input);
+
+
+	/**
+	 * @return bool Returns true iff the job supports giving feedback about the percentage done.
+	 */
+	public function supportsPercentage();
+
+
+	/**
+	 * @return int Returns 0 if !supportsPercentage and the percentage otherwise.
+	 */
+	public function getPercentage();
```

Comment:
How would i use this and/or implement this? I'm currently wondering, about this: I have a php process, that somehow instantiated the Job and called `$job->run`. Where would i use `$job->getPercentage` then? Other way round: If I would want to use this from another process: How would the two instances of the job be identified? Where, as the job, would I put my percentage info? Would everyone need to implement this on its own?